api/src/messaging/api_views.py

```diff
@@ -33,6 +39,48 @@ def get_queryset(self):
         return self.request.user.rooms.all()
 
 
+class AddRemoveMembers(APIView):
+    permission_classes = [
+        IsAuthenticated,
+        IsMember,
+    ]
+
+    @swagger_auto_schema(tags=["Messaging"])
+    def post(self, request, *args, **kwargs):
+        member_id = request.data
+        room_id = kwargs["id"]
+        try:
+            room = Room.objects.get(id=room_id)
+            if room.members.filter(id=request.user.id).exists():
+                user = CustomUser.objects.get(id=member_id)
+                room.members.add(user)
+                if room.members.filter(id=member_id):
+                    return Response({"message": "Member already exists"})
+
+                return Response({"message": "Member added to room"})
+
+            return Response(
+                {"detail": "Not found."}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        except ObjectDoesNotExist as e:
+            if e.args == ("CustomUser matching query does not exist.",):
+                return Response(
+                    {"detail": "User not found."}, status=status.HTTP_400_BAD_REQUEST
+                )
+            return Response({"detail": "Not found."}, status=status.HTTP_404_NOT_FOUND)
```

Comment:
I think it might be more descriptive to say `Room not found` here, unless there's other cases where this statement is triggered

Reply:
I have a check-in place to see if the room exists from which lines 68 and 69 are executed. Otherwise, line 71 is executed, I can change the error statuses if that would help.